automated/linux/lmp-device-register/lmp-device-register.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0-only
+# Copyright (C) 2022 Foundries.io Ltd.
+
+# shellcheck disable=SC1091
+. ../../lib/sh-test-lib
+OUTPUT="$(pwd)/output"
+RESULT_FILE="${OUTPUT}/result.txt"
+export RESULT_FILE
+TOKEN=""
+REG_UUID=""
+HSM_MODULE="/usr/lib/softhsm/libsofthsm2.so"
+
+usage() {
+    echo "Usage: $0 [-s <true|false>]" 1>&2
+    exit 1
+}
+
+while getopts "s:t:u:m:" o; do
+  case "$o" in
+    s) SKIP_INSTALL="${OPTARG}" ;;
+      t) TOKEN="${OPTARG}" ;;
+      u) REG_UUID="${OPTARG}" ;;
+      m) HSM_MODULE="${OPTARG}" ;;
+    *) usage ;;
+  esac
+done
+
+! check_root && error_msg "You need to be root to run this script."
+create_out_dir "${OUTPUT}"
+
+if [ "${SKIP_INSTALL}" = "True" ] || [ "${SKIP_INSTALL}" = "true" ]; then
+    warn_msg "Dependencies installation skipped."
+else
+      # same package name for debian and fedora
+    pkgs="uuid-runtime"
+    install_deps "${pkgs}"
+fi
+
+# quit if token is not provided
+if [ -z "${TOKEN}" ]; then
+      # exit the test
+      error_fatal "Token not provided"
+fi
+
+if [ -z "${REG_UUID}" ]; then
+      REG_UUID=$(uuidgen)
+fi
+
+if [ -z "${HSM_MODULE}" ]; then
+	HSM_MODULE="/usr/lib/softhsm/libsofthsm2.so"
```

Comment:
This is already set on line 12.
Maybe check if the file exists?
if not abort with a message.

Reply:
it's hard. When ran in LAVA it's set to empty string if not present in the test job. It's a mandatory parameter and can't be empty. So not setting it will be OK when running standalone, but not in LAVA. Any idea how to address this issue?